src/selectedtests/project_config.py

```diff
@@ -0,0 +1,208 @@
+        :param source_file_regex: The source_file_regex of the project task config.
+        :param test_file_regex: The test_file_regex of the project task config.
+        :param module: The module of the project task config.
+        :param most_recent_module_commit_analyzed: Most_recent_module_commit_analyzed of the config.
+        :param module_source_file_regex: The module_source_file_regex of the project task config.
+        :param module_test_file_regex: The module_test_file_regex of the project task config.
+        """
+        self.most_recent_project_commit_analyzed = most_recent_project_commit_analyzed
+        self.source_file_regex = source_file_regex
+        self.test_file_regex = test_file_regex
+        self.module = module
+        self.most_recent_module_commit_analyzed = most_recent_module_commit_analyzed
+        self.module_source_file_regex = module_source_file_regex
+        self.module_test_file_regex = module_test_file_regex
+
+    def update_most_recent_commits_analyzed(
+        self, most_recent_project_commit_analyzed, most_recent_module_commit_analyzed
+    ):
+        """
+        Update most recent commit fields on an instance of TaskConfig.
+
+        :param most_recent_version_analyzed: The most recent version analyzed of the project.
+        :param most_recent_module_commit_analyzed: Most_recent_module_commit_analyzed of the config.
+        """
+        self.most_recent_project_commit_analyzed = most_recent_project_commit_analyzed
+        self.most_recent_module_commit_analyzed = most_recent_module_commit_analyzed
+
+
+class ProjectConfig:
+    """Represents a project config for an Evergreen project."""
+
+    def __init__(self, project, task_config: TaskConfig, test_config: TestConfig):
+        """Init a ProjectConfig instance. Use ProjectConfig.get rather than this directly."""
+        self.project = project
+        self.task_config = task_config
+        self.test_config = test_config
+
+    @classmethod
+    def get(cls, collection: Collection, project):
+        """
+        Fetch a project config from the db. If it doesn't exist, create an empty ProjectConfig.
+
+        :param collection: The collection containing project config documents.
+        :param project: The project being fetched.
+        :return: An instance of ProjectConfig.
+        """
+        data = collection.find_one({"project": project})
+        if data:
+            return cls(
+                project,
+                TaskConfig.from_json(data.get("task_config")),
+                TestConfig.from_json(data.get("test_config")),
+            )
+        return cls(project, TaskConfig(), TestConfig())
+
+    def save(self, collection):
+        """
+        Save a ProjectConfig instance to the db collection.
+
+        :param collection: The collection containing project config documents.
+        """
+        task_config = {
```

Comment:
[nit] I would pull creating these dictionaries into the `TaskConfig` and `TestConfig` objects. So the code here would just be:
```
project_config = {
  "task_config": task_config.as_dict(),
  "test_config": test_config.as_dict(),
}
```

Maybe even pull that into it's own method to make it symmetric and here is would be:
```
collection.update({"project": self.project}, self.as_dict(), upsert=True)
```